fiftyone/core/utils.py

```diff
@@ -1144,10 +1144,317 @@ def _compute_batch_size(self):
+    scaled such that the latency between ``next()`` calls is as close as
+    possible to a specified target latency.
+
+    This class is often used in conjunction with a :class:`ProgressBar` to keep
+    the user appraised on the status of a long-running task.
+
+    Example usage::
+
+        import fiftyone.core.utils as fou
+
+        elements = range(int(1e7))
+
+        batcher = fou.LatencyDynamicBatcher(
+            elements, target_latency=0.1, max_batch_beta=2.0
+        )
+
+        for batch in batcher:
+            print("batch size: %d" % len(batch))
+
+        batcher = fou.LatencyDynamicBatcher(
+            elements,
+            target_latency=0.1,
+            max_batch_beta=2.0,
+            progress=True,
+        )
+
+        with batcher:
+            for batch in batcher:
+                print("batch size: %d" % len(batch))
+
+    Args:
+        iterable: an iterable
+        target_latency (0.2): the target latency between ``next()``
+            calls, in seconds
+        init_batch_size (1): the initial batch size to use
+        min_batch_size (1): the minimum allowed batch size
+        max_batch_size (None): an optional maximum allowed batch size
+        max_batch_beta (None): an optional lower/upper bound on the ratio
+            between successive batch sizes
+        return_views (False): whether to return each batch as a
+            :class:`fiftyone.core.view.DatasetView`. Only applicable when the
+            iterable is a :class:`fiftyone.core.collections.SampleCollection`
+        progress (False): whether to render a progress bar tracking the
+            consumption of the batches
+        total (None): the length of ``iterable``. Only applicable when
+            ``progress=True``. If not provided, it is computed via
+            ``len(iterable)``, if possible
+    """
+
+    def __init__(
+        self,
+        iterable,
+        target_latency=0.2,
+        init_batch_size=1,
+        min_batch_size=1,
+        max_batch_size=None,
+        max_batch_beta=None,
+        return_views=False,
+        progress=False,
+        total=None,
+    ):
+        super().__init__(
+            iterable,
+            target_latency,
+            init_batch_size=init_batch_size,
+            min_batch_size=min_batch_size,
+            max_batch_size=max_batch_size,
+            max_batch_beta=max_batch_beta,
+            return_views=return_views,
+            progress=progress,
+            total=total,
+        )
+
+        self._last_time = None
+
+    def _get_measurement(self):
+        current_time = timeit.default_timer()
+        time_delta = 0
+        if self._last_time is not None:
+            time_delta = current_time - self._last_time
+
+        self._last_time = current_time
+        return time_delta
+
+
+# Define this for backwards compatibility in case someone was using this
+# batcher directly
+DynamicBatcher = LatencyDynamicBatcher
+
+
+class BsonSizeDynamicBatcher(BaseDynamicBatcher):
```

Comment:
nit: elsewhere in the codebase we've historically used `BSONSizeDynamicBatcher`, ie all acronym letters are capitalized. EG `TorchCLIPModel`.

Reply:
ok sure, just a style choice. In this case, it's equally as readable.
https://stackoverflow.com/questions/2853531/how-do-you-pep-8-name-a-class-whose-name-is-an-acronym